src/views/payment-sheet-views/google-pay-view.js

```diff
@@ -0,0 +1,117 @@
+var analytics = require('../../lib/analytics');
+
+function GooglePayView() {
+  BaseView.apply(this, arguments);
+}
+
+GooglePayView.prototype = Object.create(BaseView.prototype);
+GooglePayView.prototype.constructor = GooglePayView;
+GooglePayView.ID = GooglePayView.prototype.ID = constants.paymentOptionIDs.googlePay;
+
+GooglePayView.prototype.initialize = function () {
+  var self = this;
+
+  self.googlePayConfiguration = assign({}, self.model.merchantConfiguration.googlePay);
+
+  self.model.asyncDependencyStarting();
+
+  return btGooglePay.create({client: self.client}).then(function (googlePayInstance) {
+    self.googlePayInstance = googlePayInstance;
+    self.paymentsClient = createPaymentsClient(self.client);
+  }).then(function () {
+    var buttonDiv = self.getElementById('google-pay-button');
+
+    buttonDiv.addEventListener('click', function (event) {
+      event.preventDefault();
+
+      classlist.add(self.element, 'braintree-sheet--loading');
+
+      self.tokenize().then(function () {
+        classlist.remove(self.element, 'braintree-sheet--loading');
+      });
+    });
+    self.model.asyncDependencyReady();
+  }).catch(function (err) {
+    self.model.asyncDependencyFailed({
+      view: self.ID,
+      error: new DropinError(err)
+    });
+  });
+};
+
+GooglePayView.prototype.tokenize = function () {
+  var self = this;
+  var paymentDataRequest = self.googlePayInstance.createPaymentDataRequest(self.googlePayConfiguration);
+  var rawPaymentData;
+
+  return self.paymentsClient.loadPaymentData(paymentDataRequest).then(function (paymentData) {
+    rawPaymentData = paymentData;
+    return self.googlePayInstance.parseResponse(paymentData);
+  }).then(function (tokenizePayload) {
+    tokenizePayload.rawPaymentData = rawPaymentData;
+    self.model.addPaymentMethod(tokenizePayload);
+  }).catch(function (err) {
+    var reportedError = err;
+
+    if (err.statusCode === 'DEVELOPER_ERROR') {
+      console.error(err); // eslint-disable-line no-console
+      reportedError = 'developerError';
+    } else if (err.statusCode === 'CANCELED') {
+      analytics.sendEvent(self.client, 'googlepay.loadPaymentData.canceled');
+      return;
+    } else if (err.statusCode) {
+      analytics.sendEvent(self.client, 'googlepay.loadPaymentData.failed');
+    }
+
+    self.model.reportError(reportedError);
+  });
+};
+
+GooglePayView.prototype.updateConfiguration = function (key, value) {
+  this.googlePayConfiguration[key] = value;
+};
+
+GooglePayView.isEnabled = function (options) {
+  var gatewayConfiguration = options.client.getConfiguration().gatewayConfiguration;
+
+  if (!(gatewayConfiguration.androidPay && Boolean(options.merchantConfiguration.googlePay))) {
+    return Promise.resolve(false);
+  }
+
+  return Promise.resolve().then(function () {
+    if (!global.google) {
+      return assets.loadScript(global.document.head, {
+        id: constants.GOOGLE_PAYMENT_SCRIPT_ID,
+        src: constants.GOOGLE_PAYMENT_SOURCE
```

Comment:
Do we want to mention anywhere in the docs that Drop-in will load the google pay JS for you or is that not a common thing that we document?

Reply:
I think we have the same setup for `PayPal`, and do not mention that in dropin. For dropin, basically we download all resources that we need ourselves.